apps/event-service-next/src/utils/repository.ts

```diff
@@ -0,0 +1,290 @@
+import { createClient } from '@supabase/supabase-js';
+import { getEnv } from '@hibiscus/env';
+import type { Event, EventAdmin } from '../utils/types.d.ts';
+
+const apiUrl = getEnv().Hibiscus.Supabase.apiUrl;
+const serviceKey = getEnv().Hibiscus.Supabase.serviceKey;
+
+const client = createClient(apiUrl, serviceKey, { db: { schema: 'events' } });
+
+async function getEvent(event_id: string) {
+  const { data, error } = await client
+    .from('events')
+    .select(
+      'event_id, name, start_time, end_time, location, description, event_tags (*), industry_tags (*), bp_points'
+    )
+    .eq('event_id', event_id)
+    .single();
+  if (error) throw new Error(error.message);
+
+  return {
+    eventId: data.event_id,
+    eventName: data.name,
+    startTime: data.start_time,
+    endTime: data.end_time,
+    location: data.location,
+    description: data.description,
+    eventTags: data.event_tags,
+    industryTags: data.industry_tags,
+    bpPoints: data.bp_points,
+  } as Event;
+}
+
+async function getEventAdmin(event_id: string) {
+  const { data, error } = await client
+    .from('events')
+    .select('*, event_tags (*), industry_tags (*), pinned_events (*)')
+    .eq('event_id', event_id)
+    .single();
+  if (error) throw new Error(error.message);
+
+  return {
+    eventId: data.event_id,
+    eventName: data.name,
+    startTime: data.start_time,
+    endTime: data.end_time,
+    location: data.location,
+    description: data.description,
+    eventTags: data.event_tags,
```

Comment:
eventTags and industryTags should be mapped to a list of string of the tag (same for /getEvents), currently returns entire tag object